### DISCUSSION TITLE
How to store or create user from botman

### DISCUSSION THREAD

**Discussion by donmbelembe** [botman/botman#1360] [General]
Hello, I'm trying to know if there is best practice to create a user the first time someone text you.
looks like there is already a thread on the forum, but the forum seems down. https://botman.io/forum/d/40-how-to-store-user-information-to-mysql-database

**Comment by filippotoso** [comment]
Just create a table with two fields:  platform and platform_id

The platform field will be filled with something like webchat, telegram and so on depending on the driver you are using
The platform_id will contain the bot user id (https://botman.io/2.0/user-information#retrieving-user-id)

Then you can user a middleware (https://botman.io/2.0/middleware) to updateOrCreate() a user from those two fields.
